### ISSUE TITLE
Don't use smart constructors for visible types

### ISSUE THREAD

**Issue by mlutze** [flix/flix#4370]
E.g. `def f(x: Int32): Int32 \ { ef - ef }` should not be reduced to `def f(x: Int32): Int32 \ { }`, since `ef` is visible. There are two reasons:
- Maintaining specific type location information for LSP
- Maintaining specific type location information for Redundancy

**Comment by magnus-madsen** [comment]
I guess this is relevant to the current refactorings.

**Comment by mlutze** [comment]
Yes. I wish we had a way to enforce this at the type level, since we do a lot of manipulation and it is easy to get an errant `minimize` in there
